src/coreclr/nativeaot/Runtime/thread.cpp

```diff
@@ -547,7 +549,14 @@ void Thread::GcScanRootsWorker(void * pfnEnumCallback, void * pvCallbackData, St
 
 EXTERN_C void FASTCALL RhpSuspendRedirected();
 
-#ifndef TARGET_ARM64
+#if defined(TARGET_ARM64) || defined(TARGET_UNIX)
+EXTERN_C void FASTCALL RhpGcProbeHijack();
+
+static void* NormalHijackTargets[1] =
+{
+    reinterpret_cast<void*>(RhpGcProbeHijack)
+};
+#else // TARGET_ARM64 || TARGET_UNIX
 EXTERN_C void FASTCALL RhpGcProbeHijackScalar();
```

Comment:
It would be nice to switch the other platforms to the one-and-only hijack target plan (in separate PR).

Reply:
Yes. It occured to me that using multiple probe targets on win-x64 as a way to pass return kind to the probe is not very rewarding optimization.  
Perf of suspension code would be gated by how many syscalls we make and how many iterative attempts we do, so benefits here would be fairly small in comparison.